uber/utils.py

```diff
@@ -225,8 +250,47 @@ def __init__(self, targets=(), amount=None, description=None, email=''):
         self.description = description or self.names
         self.email = self.models[0].email if self.targets and self.models[0].email else email
 
-    @staticmethod
-    def to_sessionized(m):
+    @classproperty
+    def paid_preregs(cls):
+        return cherrypy.session.setdefault('paid_preregs', [])
+
+    @classproperty
+    def unpaid_preregs(cls):
+        return cherrypy.session.setdefault('unpaid_preregs', OrderedDict())
+
+    @classmethod
+    def get_unpaid_promo_code_uses_count(cls, id, already_counted_attendee_ids=None):
+        attendees_with_promo_code = set()
+        if already_counted_attendee_ids:
+            attendees_with_promo_code.update(listify(already_counted_attendee_ids))
+
+        promo_code_count = 0
+
+        targets = [t for t in cls.unpaid_preregs.values() if '_model' in t]
+        for target in targets:
+            if target['_model'] == 'Attendee':
+                if target.get('id') not in attendees_with_promo_code \
+                        and target.get('promo_code') \
+                        and target['promo_code'].get('id') == id:
+                    attendees_with_promo_code.add(target.get('id'))
+                    promo_code_count += 1
+
+            elif target['_model'] == 'Group':
```

Comment:
Wait, you disabled using promo codes when registering a group, right? How is this block ever used?